FxCopAnalyzer fixes for Wox.Test

avneet-kr opened the pull request:
## Summary of the Pull Request

This PR includes fixes for FxCop errors for Wox.Test

## PR Checklist
* [x] Applies to #5129 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [x] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

- Enabled FxCopAnalyzer for Wox.Test
- Added InvarientCulture info (CA1305: Specify IFormatProvider)
- Removed underscore from method names (CA1707: Identifiers should not contain underscores)
- Marked members as static (CA1822: Mark members as static)
- Removed initialization for DummyTestClass::IsFunctionCalled (CA1805: Do not initialize unnecessarily)
- Replaced Count with Any (CA1827: Do not use Count/LongCount when Any can be used)
- Changed WoxTest::DummyTestClass to private (CA1034: Nested types should not be visible)
- Suppressed warnings for uninstantiated internal classes used as TestFixtures (CA1812: Avoid uninstantiated internal classes)
    - Warnings were suppressed per this issue- https://github.com/dotnet/roslyn-analyzers/issues/1830 

## Validation Steps Performed

All changes were validated by building and running unit tests

avneet-kr committed:
Merge pull request #1 from microsoft/master

Pulling in changes from microsoft/PowerToys

avneet-kr committed:
Added InvarientCulture info (CA1305: Specify IFormatProvider)

avneet-kr committed:
Removed underscore from method names (CA1707: Identifiers should not contain underscores)

avneet-kr committed:
Marked members as static (CA1822: Mark members as static)

avneet-kr committed:
Removed initialization for DummyTestClass::IsFunctionCalled (CA1805: Do not initialize unnecessarily)

avneet-kr committed:
Replaced Count with Any (CA1827: Do not use Count/LongCount when Any can be used)

avneet-kr committed:
Changed WoxTest::DummyTestClass to private (CA1034: Nested types should not be visible)

avneet-kr committed:
Suppressed warnings for uninstantiated internal classes used as TestFixtures (CA1812: Avoid uninstantiated internal classes)

avneet-kr committed:
Enabled FxCopAnalyzer for Wox.Test

somil55 commented on src/modules/launcher/Wox.Test/FuzzyMatcherTest.cs:
```diff
@@ -228,7 +230,7 @@ public void WhenMultipleResults_ExactMatchingResult_ShouldHaveGreatestScore(stri
         [TestCase("Windows Terminal", "Windows_Terminal", "term")]
         [TestCase("Windows Terminal", "WindowsTerminal", "term")]
-        public void FuzzyMatchingScore_ShouldBeHigher_WhenPreceedingCharacterIsSpace(string firstCompareStr, string secondCompareStr, string query)
+        public void FuzzyMatchingScoreShouldBeHigherWhenPreceedingCharacterIsSpace(string firstCompareStr, string secondCompareStr, string query)
```
@ryanbodrug-microsoft FxCop here breaks our test naming convention. Do you think we should keep our convention and instead suppress this error for the entire project ?

ryanbodrug-microsoft replied:
I think the naming is the same.  It's just removing the underscores.  I'm okay with this change as is, but I'm used to reading the "Should -> When" syntax.  My personal opinion would be to conform with broader conventions as those imposed by fxcop.

somil55 commented on src/modules/launcher/Wox.Test/Plugins/FolderPluginTest.cs:
```diff
@@ -10,6 +10,7 @@
 
 namespace Wox.Test.Plugins
 {
+#pragma warning disable CA1812 // Suppressing warning for uninstantiated internal class
```
I think we should place this error in a global suppression file for better code readability. You can get more info here: https://docs.microsoft.com/en-us/visualstudio/code-quality/in-source-suppression-overview?view=vs-2019#global-level-suppressions

avneet-kr replied:
Thanks @somil55. I've updated the PR and added the suppression in GlobalSuppressions.cs instead of inline.

avneet-kr committed:
Added suppression message for CA1812 in GlobalSuppressions.cs and removed inline supression messages (CA1812: Avoid uninstantiated internal classes)

ryanbodrug-microsoft commented on src/modules/launcher/Wox.Test/QueryBuilderTest.cs:
```diff
@@ -194,8 +197,9 @@ public void QueryBuilder_ShouldSetTermsCorrently_WhenCalled()
-            Assert.IsTrue(firstQuery.Terms[0].Equals("cd") && firstQuery.Terms[1].Equals("efgh") && firstQuery.Terms.Length == 2);
-            Assert.IsTrue(secondQuery.Terms[0].Equals("efgh") && secondQuery.Terms.Length == 1);
+            // Using InvariantCulture since this is a command line arg
+            Assert.IsTrue(firstQuery.Terms[0].Equals("cd", StringComparison.InvariantCulture) && firstQuery.Terms[1].Equals("efgh", StringComparison.InvariantCulture) && firstQuery.Terms.Length == 2);
```
I just want to make sure the use of InvariantCulture is intentional here.  This will treat different characters with the same linguistic meaning 'æ' and 'ae' as equal.    I don't think it makes a difference here but we should be intentional with the parameters we are setting.

avneet-kr replied:
Thanks @ryanbodrug-microsoft. I've updated the PR to use Ordinal in place of InvariantCulture.

ryanbodrug-microsoft commented on src/codeAnalysis/GlobalSuppressions.cs:
```diff
@@ -45,3 +45,6 @@
 [assembly: SuppressMessage("Microsoft.Globalization", "CA1303:Do not pass literals as localized parameters", MessageId = "System.Windows.Documents.InlineCollection.Add(System.String)", Scope = "member", Target = "Microsoft.Templates.UI.Extensions.TextBlockExtensions.#OnSequentialFlowStepChanged(System.Windows.DependencyObject,System.Windows.DependencyPropertyChangedEventArgs)", Justification = "N ...
+
+// FxCop warning suppression for uninstantiated TestFixture classes
+[assembly: SuppressMessage("Microsoft.Performance", "CA1812: Avoid uninstantiated internal classes", Scope = "module", Justification = "The classes are used as TestFixture")]
```
I don't understand why we would need this.  Can you point me to the offending line of code?

avneet-kr replied:
This change was made to address @somil55 's comments above to add the suppression globally for better code readability,, referring to https://github.com/microsoft/PowerToys/pull/7312/commits/685e9034f3a022a5a075e1f12117a5a567eb1aa3.

somil55 replied:
@ryanbodrug-microsoft CA1812 will be thrown for every file in the test project. This is mentioned here: https://github.com/dotnet/roslyn-analyzers/issues/1830. So rather than handling it manually in each file, it might be better to handle it in a global suppression file. The classes instantiated in this project are only for test purposes, so this should not affect anything else.

ryanbodrug-microsoft replied:
Sounds good thanks for the info.  Can we update theJustification parameter to include the explanation from divyansh "CA1812 will be thrown for every file in the test project. This is mentioned here: dotnet/roslyn-analyzers#1830"

avneet-kr committed:
Changed InvariantCulture to Ordinal where appropriate

avneet-kr committed:
Addressing comments: Updated justification for CA1812 in GlobalSuppressions.cs (Avoid uninstantiated internal classes)

avneet-kr commented:
PR has been updated to address comments:
- Changed InvariantCulture to Ordinal where appropriate
- Added suppression message for CA1812 in GlobalSuppressions.cs and removed inline supression messages (CA1812: Avoid uninstantiated internal classes)
- Updated justification for CA1812 in GlobalSuppressions.cs (Avoid uninstantiated internal classes)

avneet-kr commented:
/azp run
